extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -0,0 +1,361 @@
+		this.splitView.addItem(this.diffEditor);
+		this.splitView.setLayout({
+			orientation: 'vertical',
+			splitViewHeight: 800
+		});
+
+		this.flexModel.removeItem(this.loader);
+		this.switchButton.enabled = true;
+		this.compareButton.enabled = true;
+
+		if (this.comparisonResult.differences.length > 0) {
+			this.flexModel.addItem(this.splitView);
+
+			// only enable generate script button if the target is a db
+			if (this.targetEndpointInfo.endpointType === azdata.SchemaCompareEndpointType.database) {
+				this.generateScriptButton.enabled = true;
+			} else {
+				this.generateScriptButton.title = localize('schemaCompare.generateScriptButtonDisabledTitle', 'Generate script is enabled when the target is a database');
+			}
+		} else {
+			this.flexModel.addItem(this.noDifferencesLabel, { CSSStyles: { 'margin': 'auto' } });
+		}
+
+		let sourceText = '';
+		let targetText = '';
+		this.differencesTable.onRowSelected(() => {
+			let difference = this.comparisonResult.differences[this.differencesTable.selectedRows[0]];
+			if (difference !== undefined) {
+				sourceText = difference.sourceScript === null ? '\n' : this.getAggregatedScript(difference, true);
+				targetText = difference.targetScript === null ? '\n' : this.getAggregatedScript(difference, false);
+
+				this.diffEditor.updateProperties({
+					contentLeft: sourceText,
+					contentRight: targetText,
+					title: localize('schemaCompare.ObjectDefinitionsTitle', 'Object Definitions')
+				});
+			}
+		});
+	}
+
+	private getAllDifferences(differences: azdata.DiffEntry[]): string[][] {
+		let data = [];
+		if (differences) {
+			differences.forEach(difference => {
+				if (difference.differenceType === azdata.SchemaDifferenceType.Object) {
+					if (difference.sourceValue !== null || difference.targetValue !== null) {
+						data.push([difference.name, difference.sourceValue, this.SchemaCompareActionMap[difference.updateAction], difference.targetValue]);
+					}
+				}
+			});
+		}
+
+		return data;
+	}
+
+	private getAggregatedScript(diffEntry: azdata.DiffEntry, getSourceScript: boolean): string {
+		let script = '';
+		if (diffEntry !== null) {
+			script += getSourceScript ? diffEntry.sourceScript : diffEntry.targetScript;
+			diffEntry.children.forEach(child => {
+				let childScript = this.getAggregatedScript(child, getSourceScript);
+				if (childScript !== 'null') {
+					script += childScript;
+				}
+			});
+		}
+		return script;
+	}
+
+	private reExecute(): void {
+		this.flexModel.removeItem(this.splitView);
+		this.flexModel.removeItem(this.noDifferencesLabel);
+		this.flexModel.addItem(this.loader, { CSSStyles: { 'margin-top': '30px' } });
+		this.diffEditor.updateProperties({
+			contentLeft: '\n',
+			contentRight: '\n'
+		});
+		this.differencesTable.selectedRows = null;
+		this.resetButtons();
+		this.execute();
+	}
+
+	private createCompareButton(view: azdata.ModelView): void {
+		let runIcon = path.join(__dirname, '.', 'media', 'compare.svg');
```

Comment:
I am afraid we don't have a way to specify theme specific icons for this component, so please ask Andrew to give you an icon that works for both Dark and Light themes.

Reply:
this applies to all the icons in the toolbar

Reply:
@alanrenmsft The icon I showed you were not right - this is how they look (still not good bit better than what was shown in my screen)
![image](https://user-images.githubusercontent.com/46980425/56250649-17df8a00-6065-11e9-9bc2-5b30ff9806d0.png)